pyecsca/sca/stacked_traces/stacked_traces.py

```diff
@@ -0,0 +1,206 @@
+from numba import cuda
+import numpy as np
+from public import public
+from typing import Any, Mapping, MutableSequence, Tuple, Union
+from math import sqrt
+
+from pyecsca.sca.trace.trace import CombinedTrace
+
+
+@public
+class StackedTraces:
+    """Samples of multiple traces and metadata"""
+
+    meta: Mapping[str, Any]
+    samples: np.ndarray
+
+    def __init__(
+            self, samples: np.ndarray,
+            meta: Mapping[str, Any] = None) -> None:
+        if meta is None:
+            meta = {}
+        self.meta = meta
+        self.samples = samples
+
+    @classmethod
+    def fromarray(cls, traces: MutableSequence[np.ndarray],
+                  meta: Mapping[str, Any] = None) -> 'StackedTraces':
+        min_samples = min(map(len, traces))
+        for i, t in enumerate(traces):
+            traces[i] = t[:min_samples]
+        stacked = np.stack(traces)
+        return cls(stacked, meta)
+
+    @classmethod
+    def fromtraceset(cls, traceset) -> 'StackedTraces':
+        traces = [t.samples for t in traceset]
+        return cls.fromarray(traces)
+
+    def __len__(self):
+        return self.traces.shape[0]
+
+    def __getitem__(self, index):
+        return self.traces
+
+    def __iter__(self):
+        yield from self.traces
+
+
+@public
+class GPUTraceManager:
+    TPB = Union[int, Tuple[int, ...]]
+    BPG = Union[int, Tuple[int, ...]]
+    Samples = cuda.devicearray.DeviceNDArray
+    Output = cuda.devicearray.DeviceNDArray
+    CudaCTX = Tuple[Samples, Tuple[Output, ...], BPG]
```

Comment:
This type of type hint is weird and mostly unused (see below: `tpb: int`). Also mypy complains about them: https://github.com/J08nY/pyecsca/runs/5023232820?check_suite_focus=true and points to https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases.

I suggest to fix it according to the mypy guide.

Reply:
Also, these values lack documentation, both as types and as arguments to the methods, which needs to be added.